spark/src/main/scala/org/apache/spark/sql/delta/commands/alterDeltaTableCommands.scala

```diff
@@ -182,6 +182,63 @@ case class AlterTableUnsetPropertiesDeltaCommand(
   }
 }
 
+/**
+ * A command that removes an existing feature from the table. The feature needs to implement the
+ * [[RemovableFeature]] trait.
+ *
+ * The syntax of the command is:
+ * {{{
+ *   ALTER TABLE t DROP FEATURE f
+ * }}}
+ */
+case class AlterTableDropFeatureDeltaCommand(
+    table: DeltaTableV2,
+    featureName: String)
+  extends LeafRunnableCommand with AlterDeltaTableCommand with IgnoreCachedData {
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val deltaLog = table.deltaLog
+    recordDeltaOperation(deltaLog, "delta.ddl.alter.dropFeature") {
+      // This guard is only temporary while the remove feature is in development.
+      require(sparkSession.conf.get(DeltaSQLConf.TABLE_FEATURE_REMOVAL_ENABLED))
+
+      val removableFeature = TableFeature.featureNameToFeature(featureName) match {
+        case Some(feature: RemovableFeature) => feature
+        case Some(_) => throw DeltaErrors.dropTableFeatureNonRemovableFeature(featureName)
+        case None => throw DeltaErrors.dropTableFeatureFeatureNotSupportedByClient(featureName)
+      }
+
+      if (!table.snapshot.protocol.readerAndWriterFeatureNames.contains(featureName)) {
+        throw DeltaErrors.dropTableFeatureFeatureNotSupportedByProtocol(featureName)
+      }
```

Comment:
Why is this needed? Until we support (at least some) reader-writer features, none of those will impement the `RemovableFeature` trait and we should have already thrown at L267 above. 

If some of the code below is known-broken for reader-writer features, then this should be an assertion with a big doc comment explaining why the specific check is needed.

Reply:
I am afraid this is misleading but `readerAndWriterFeatureNames` includes both writer and reader+writer features. A more appropriate name would be readerWriterAndWriterFeatureNames. I added a comment in the code to disambiguate this.

Reply:
How about just `allFeatureNames`? :)